diff -urN a/ext/intl/breakiterator/codepointiterator_internal.cpp b/ext/intl/breakiterator/codepointiterator_internal.cpp
--- a/ext/intl/breakiterator/codepointiterator_internal.cpp	2022-02-08 09:00:24.762166166 +0100
+++ b/ext/intl/breakiterator/codepointiterator_internal.cpp	2022-02-08 09:01:12.916739745 +0100
@@ -72,7 +72,7 @@
 	clearCurrentCharIter();
 }
 
-UBool CodePointBreakIterator::operator==(const BreakIterator& that) const
+bool CodePointBreakIterator::operator==(const BreakIterator& that) const
 {
 	if (typeid(*this) != typeid(that)) {
 		return FALSE;
diff -urN a/ext/intl/breakiterator/codepointiterator_internal.h b/ext/intl/breakiterator/codepointiterator_internal.h
--- a/ext/intl/breakiterator/codepointiterator_internal.h	2022-02-08 09:00:24.762166166 +0100
+++ b/ext/intl/breakiterator/codepointiterator_internal.h	2022-02-08 09:01:28.092605358 +0100
@@ -36,7 +36,7 @@
 
 		virtual ~CodePointBreakIterator();
 
-		virtual UBool operator==(const BreakIterator& that) const;
+		virtual bool operator==(const BreakIterator& that) const;
 
 		virtual CodePointBreakIterator* clone(void) const;
 
