Upstream-Status: Inappropriate [cross compile specific]

diff -durpN php-7.4.11.orig/configure.ac php-7.4.11/configure.ac
--- php-7.4.11.orig/configure.ac	2019-03-27 06:32:21.228353553 -0400
+++ php-7.4.11/configure.ac	2019-03-27 06:46:07.430549722 -0400
@@ -1317,7 +1317,7 @@
 
 dnl SOURCE_DATE_EPOCH for reproducible builds
 dnl https://reproducible-builds.org/specs/source-date-epoch/
-PHP_BUILD_DATE=`date --utc --date="@${SOURCE_DATE_EPOCH:-$(date +%s)}" +%Y-%m-%d 2>/dev/null`
+PHP_BUILD_DATE=`date --utc --date="@${SOURCE_DATE_EPOCH:-$(date +%s -r $srcdir/NEWS)}" +%Y-%m-%d 2>/dev/null`
 if test $? -ne 0 ; then
 PHP_BUILD_DATE=`date -u +%Y-%m-%d`
 fi
