Upstream-Status: Inappropriate [oe specific]

--- php-7.4.26/ext/openssl/openssl.c~	2021-11-19 06:16:13.000000000 +0000
+++ php-7.4.26/ext/openssl/openssl.c	2021-11-19 06:20:14.666373542 +0000
@@ -65,6 +65,7 @@
 
 /* Common */
 #include <time.h>
+#include <stdbool.h> /* for declaration of "true" */
 
 #if (defined(PHP_WIN32) && defined(_MSC_VER) && _MSC_VER >= 1900)
 #define timezone _timezone	/* timezone is called _timezone in LibC */
