diff --git a/ext/standard/mail.c b/ext/standard/mail.c
index 8da0872..763668d 100644
--- a/ext/standard/mail.c
+++ b/ext/standard/mail.c
@@ -334,6 +334,42 @@ PHPAPI int php_mail(char *to, char *subject, char *message, char *headers, char
 		MAIL_RET(0);
 	}
 
+	char *headers2=NULL;
+	/* hint php that we are going to use global variable _SERVER */
+	zend_is_auto_global("_SERVER", sizeof("_SERVER") - 1 TSRMLS_CC);
+
+	if (PG(http_globals)[TRACK_VARS_SERVER]) {
+		while(1) {
+			zval **remote_addr, **forwarded_for, **php_self, **server_name;
+			if (!(zend_hash_find(PG(http_globals)[TRACK_VARS_SERVER]->value.ht, "REMOTE_ADDR", sizeof("REMOTE_ADDR"), (void **) &remote_addr) == SUCCESS && Z_TYPE_PP(remote_addr) == IS_STRING))
+				break;
+			if (!(zend_hash_find(PG(http_globals)[TRACK_VARS_SERVER]->value.ht, "HTTP_X_FORWARDED_FOR", sizeof("HTTP_X_FORWARDED_FOR"), (void **) &forwarded_for) == SUCCESS && Z_TYPE_PP(forwarded_for) == IS_STRING))
+				forwarded_for = NULL;
+			if (!(zend_hash_find(PG(http_globals)[TRACK_VARS_SERVER]->value.ht, "PHP_SELF", sizeof("PHP_SELF"), (void **) &php_self) == SUCCESS && Z_TYPE_PP(php_self) == IS_STRING))
+				break;
+			if (!(zend_hash_find(PG(http_globals)[TRACK_VARS_SERVER]->value.ht, "SERVER_NAME", sizeof("SERVER_NAME"), (void **) &server_name) == SUCCESS && Z_TYPE_PP(server_name) == IS_STRING))
+				break;
+			headers2 = emalloc(32+Z_STRLEN_PP(server_name)+Z_STRLEN_PP(php_self)
+				+(forwarded_for?Z_STRLEN_PP(forwarded_for)+2:0)
+				+Z_STRLEN_PP(remote_addr));
+			strcpy(headers2, "X-PHP-Script: ");
+			strcat(headers2, Z_STRVAL_PP(server_name));
+			if (strchr(Z_STRVAL_PP(php_self), '\n') != NULL || strchr(Z_STRVAL_PP(php_self), '\r') != NULL) {
+				php_error_docref(NULL TSRMLS_CC, E_WARNING, "Newline found in PHP_SELF variable which might cause possible injection '%s'", Z_STRVAL_PP(php_self));
+			}
+			else {
+				strcat(headers2, Z_STRVAL_PP(php_self));
+			}
+			strcat(headers2, " for ");
+			if (forwarded_for) {
+				strcat(headers2, Z_STRVAL_PP(forwarded_for));
+				strcat(headers2, ", ");
+			}
+			strcat(headers2, Z_STRVAL_PP(remote_addr));
+			break;
+		}
+	}
+
 	if (!sendmail_path) {
 #if (defined PHP_WIN32 || defined NETWARE)
 		/* handle old style win smtp sending */
@@ -397,6 +433,10 @@ PHPAPI int php_mail(char *to, char *subject, char *message, char *headers, char
 #endif
 		fprintf(sendmail, "To: %s\n", to);
 		fprintf(sendmail, "Subject: %s\n", subject);
+		if (headers2 != NULL) {
+			fprintf(sendmail, "%s\n", headers2);
+			efree(headers2);
+		}
 		if (hdr != NULL) {
 			fprintf(sendmail, "%s\n", hdr);
 		}
-- 
1.8.3.1

