--- a/src/Makefile.am	2024-02-01 18:24:44.942894636 +0100
+++ b/src/Makefile.am	2024-02-01 18:25:04.214464617 +0100
@@ -26,10 +26,9 @@
 	@# wheel-based installation with .dist-info.
 	@# This needs to work on RHEL8 up through modern Fedora, offline, with
 	@# system packages available to the build.
-	python3 -m pip install --no-index --force-reinstall --root='$(DESTDIR)/' --prefix='$(prefix)' \
+	python3 -m pip install --no-index --root='$(DESTDIR)/' --prefix='$(prefix)' \
 		"$$(python3 '$(srcdir)'/src/build_backend.py --wheel '$(srcdir)' tmp/wheel)"
 	mkdir -p $(DESTDIR)$(libexecdir)
-	mv -t $(DESTDIR)$(libexecdir) $(DESTDIR)$(bindir)/cockpit-askpass
 
 UNINSTALL_LOCAL_TARGETS += uninstall-python
 uninstall-python:
