diff --git a/src/protocols/rdp/channels/rdpdr/rdpdr-printer.c b/src/protocols/rdp/channels/rdpdr/rdpdr-printer.c
index d90116f..3abc5e3 100644
--- a/src/protocols/rdp/channels/rdpdr/rdpdr-printer.c
+++ b/src/protocols/rdp/channels/rdpdr/rdpdr-printer.c
@@ -55,6 +55,19 @@ void guac_rdpdr_process_print_job_create(guac_rdp_common_svc* svc,
 
 }
 
+void guac_rdpdr_process_print_job_read(guac_rdp_common_svc* svc,
+        guac_rdpdr_device* device, guac_rdpdr_iorequest* iorequest,
+        wStream* input_stream) {
+
+    /* Respond with not_supported */
+    wStream* output_stream = guac_rdpdr_new_io_completion(device,
+            iorequest->completion_id, STATUS_NOT_SUPPORTED, 4);
+
+    Stream_Write_UINT32(output_stream, 0); /* fileId */
+    guac_rdp_common_svc_write(svc, output_stream);
+
+}
+
 void guac_rdpdr_process_print_job_write(guac_rdp_common_svc* svc,
         guac_rdpdr_device* device, guac_rdpdr_iorequest* iorequest,
         wStream* input_stream) {
@@ -128,19 +141,27 @@ void guac_rdpdr_device_printer_iorequest_handler(guac_rdp_common_svc* svc,
 
         /* Print job create */
         case IRP_MJ_CREATE:
+            guac_client_log(svc->client, GUAC_LOG_INFO, "printer create");
             guac_rdpdr_process_print_job_create(svc, device, iorequest, input_stream);
             break;
 
         /* Printer job write */
         case IRP_MJ_WRITE:
+            guac_client_log(svc->client, GUAC_LOG_DEBUG, "printer write");
             guac_rdpdr_process_print_job_write(svc, device, iorequest, input_stream);
             break;
 
         /* Printer job close */
         case IRP_MJ_CLOSE:
+            guac_client_log(svc->client, GUAC_LOG_INFO, "printer close");
             guac_rdpdr_process_print_job_close(svc, device, iorequest, input_stream);
             break;
 
+        case IRP_MJ_READ:
+            guac_client_log(svc->client, GUAC_LOG_INFO, "printer read");
+            guac_rdpdr_process_print_job_read(svc, device, iorequest, input_stream);
+            break;
+
         /* Log unknown */
         default:
             guac_client_log(svc->client, GUAC_LOG_ERROR, "Unknown printer "
diff --git a/src/protocols/rdp/channels/rdpdr/rdpdr-printer.h b/src/protocols/rdp/channels/rdpdr/rdpdr-printer.h
index a6865d8..fd78bdf 100644
--- a/src/protocols/rdp/channels/rdpdr/rdpdr-printer.h
+++ b/src/protocols/rdp/channels/rdpdr/rdpdr-printer.h
@@ -60,6 +60,12 @@ guac_rdpdr_device_iorequest_handler guac_rdpdr_process_print_job_create;
 guac_rdpdr_device_iorequest_handler guac_rdpdr_process_print_job_write;
 
 /**
+ * I/O request handler which processes a request to read data to an existing
+ * print job.
+ */
+guac_rdpdr_device_iorequest_handler guac_rdpdr_process_print_job_read;
+
+/**
  * I/O request handler which processes a request to close an existing print
  * job.
  */
