From 3a34c9566ace6d5b84fcc5f41a2e44064a718399 Mon Sep 17 00:00:00 2001
From: Joe Watkins <krakjoe@php.net>
Date: Sat, 19 Nov 2016 05:13:16 +0000
Subject: [PATCH] disable cache gracefully in cli krakjoe/pthreads#639

---
 php_apc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/php_apc.c b/php_apc.c
index 60b16172..a1cf8c24 100644
--- a/php_apc.c
+++ b/php_apc.c
@@ -269,7 +269,7 @@ static PHP_MINIT_FUNCTION(apcu)
 	
     /* Disable APC in cli mode unless overridden by apc.enable_cli */
     if (!APCG(enable_cli) && !strcmp(sapi_module.name, "cli")) {
-        APCG(enabled) = 0;
+		zend_alter_ini_entry("apc.enabled", sizeof("apc.enabled"), "0", sizeof("0"), PHP_INI_SYSTEM, PHP_INI_STAGE_STARTUP);
     }
 
 	/* only run initialization if APC is enabled */
